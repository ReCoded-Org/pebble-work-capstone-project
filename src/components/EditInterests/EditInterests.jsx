import { useState } from "react";

import Button from "@/components/Button";

import axios from "@/api/axios";
const EditInterests = ({ beforeClick, afterClick }) => {
    const labels = [
        "No Poverty",
        "Zero Hunger",
        "Good Health And Well-Being",
        "Quality Education",
        "Gender Equality",
        "Clean Water And Sanitation",
        "Affordable And Clean Energy",
        "Decent Work And Economic Growth",
        "Industry Innovation And Infrastructure",
        "Reduced Inequalities",
        "Sustainable Cities And Communities",
        "Responsible Consumption And Production",
        "Climate Action",
        "Life Below Water",
        "Life On Land",
        "Peace And Justice And Strong Institutions",
        "Partnership For The Goals",
    ];
    const [selected, setSelected] = useState([]);

    const handleClick = (label) => {
        setSelected([...selected, label]);
        if (selected.includes(label)) {
            setSelected(selected.filter((item) => item !== label));
        }
    };

    async function submitinterests() {
        var bodyFormData = new FormData();
        for (const a of selected) {
            bodyFormData.append("interests[]", a);
        }
        try {
            const response = await axios({
                method: "put",
                url: "/api/user/",
                data: bodyFormData,
                headers: { "Content-Type": "multipart/form-data" },
                withCredentials: true,
            });
            //const data = await response.json()
            console.log(response);
            // router.push(response.data._id);
        } catch (err) {
            //console.log(err);
        }
    }
    return (
        <div className='m-4 md:m-8 '>
            <div className='flex flex-wrap items-center justify-center overflow-hidden lg:ml-8 lg:mr-8 lg:flex-row lg:flex-wrap lg:items-center lg:justify-center'>
                {labels.map((label) => (
                    <div key={label} className='m-1 md:m-2 lg:m-3'>
                        <Button
                            label={label}
                            bgColor={
                                selected.includes(label)
                                    ? "bg-primary-200"
                                    : "bg-white"
                            }
                            fontWeight='font-medium'
                            width='w-auto'
                            height='h-7 md:h-12 lg:h-12'
                            fontSize='text-[9px] md:text-xl lg:text-lg'
                            border='border lg:border-2'
                            customStyle={
                                selected.includes(label)
                                    ? `${afterClick}`
                                    : `text-black ${beforeClick}`
                            }
                            onClick={() => handleClick(label)}
                        />
                    </div>
                ))}
            </div>
            <div className='flex justify-end pr-8  '>
                <Button
                    label='Change my interest'
                    bgColor='bg-primary-200'
                    textColor='text-white'
                    fontWeight='font-medium'
                    width='w-40 md:w-40 lg:w-52'
                    height='h-8 md:h-12 lg:h-12'
                    borderColor=''
                    fontSize='text-[11px] md:text-xl lg:text-xl'
                    border=''
                    customStyle='flex items-center justify-center'
                    onClick={submitinterests}
                />
            </div>
        </div>
    );
};

export default EditInterests;
