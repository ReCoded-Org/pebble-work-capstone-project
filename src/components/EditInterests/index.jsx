import EditInterests from "./EditInterests";
export default EditInterests;
