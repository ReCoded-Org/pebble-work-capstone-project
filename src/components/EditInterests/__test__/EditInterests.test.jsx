import renderer from "react-test-renderer";

import EditInterests from "../EditInterests";

it("renders correctly", () => {
    const tree = renderer.create(<EditInterests />).toJSON();
    expect(tree).toMatchSnapshot();
});
