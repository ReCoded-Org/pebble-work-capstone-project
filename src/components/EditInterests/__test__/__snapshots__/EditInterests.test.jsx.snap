// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="m-4 mt-12 md:m-12"
>
  <h2
    className="text-md m-4 font-medium md:m-6 md:text-2xl lg:mt-16 lg:ml-12 lg:text-3xl"
  >
    Your Interests
  </h2>
  <div
    className="flex flex-wrap items-center justify-center overflow-hidden lg:ml-8 lg:flex-row lg:flex-wrap lg:items-start lg:justify-start"
  >
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        No Poverty
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Zero Hunger
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Good Health and Well-Being
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Quality Education
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Gender Equality
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Clean Water and Sanitation
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Affordable and Clean Energy
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Decent Work and Economic Growth
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Industry, Innovation and Infrastructure
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Reduced Inequalities
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Sustainable Cities and Communities
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Responsible Consumption and Production
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Climate Action
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Life Below Water
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Life On Land
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Peace, Justice and Strong Institutions
      </button>
    </div>
    <div
      className="m-1 md:m-2 lg:m-3"
    >
      <button
        className="mt-1 mb-1 w-auto h-7 md:h-12 lg:h-12 text-[9px] md:text-xl lg:text-lg bg-white border lg:border-2 undefined hover:undefined text-white font-medium  py-1  px-2 undefined hover:undefined rounded  hover:shadow-lg undefined"
        onClick={[Function]}
        type="button"
      >
        Partnerships to achieve the Goal
      </button>
    </div>
  </div>
</div>
`;
